Fix panic

How to reproduce:
```
$ minikube delete; minikube start
$ ./hack/docker/voyager/setup.sh; ./hack/docker/voyager/setup.sh push
$ env APPSCODE_ENV=dev ./hack/deploy/voyager.sh --provider=minikube
$ ./hack/make.py test minikube --selfhosted-operator
```
https://github.com/appscode/voyager/blob/master/pkg/haproxy/template/template.go#L45

```
I0512 03:40:06.540268       1 parser.go:124] [4507ef1a-0309-498a-b8f8-37bb410caaf0] Found target port 9090 for service e2e-test-server-wyxhc
I0512 03:40:06.541277       1 ingress_crds.go:95] Sync/Add/Update for engress test-5woan/e2e-test-7fdjuq
I0512 03:40:06.541850       1 parser.go:29] [9593cc2f-292c-4238-871d-3780086cc8eb] [getting endpoints for  test-5woan e2e-test-server-wyxhc port {0 9090 }]
I0512 03:40:06.541907       1 parser.go:42] [9593cc2f-292c-4238-871d-3780086cc8eb] [looking for services in namespace test-5woan with name e2e-test-server-wyxhc]
E0512 03:40:06.544455       1 runtime.go:66] Observed a panic: "port expected to be > 0, found 0" (port expected to be > 0, found 0)
/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/text/template/exec.go:146
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/go/src/github.com/appscode/voyager/pkg/haproxy/template/template.go:45
/usr/local/go/src/runtime/asm_amd64.s:510
/usr/local/go/src/reflect/value.go:434
/usr/local/go/src/reflect/value.go:302
/usr/local/go/src/text/template/exec.go:670
/usr/local/go/src/text/template/exec.go:538
/usr/local/go/src/text/template/exec.go:435
/usr/local/go/src/text/template/exec.go:408
/usr/local/go/src/text/template/exec.go:438
/usr/local/go/src/text/template/exec.go:408
/usr/local/go/src/text/template/exec.go:263
/usr/local/go/src/text/template/exec.go:253
/usr/local/go/src/text/template/exec.go:242
/usr/local/go/src/text/template/exec.go:329
/usr/local/go/src/text/template/exec.go:338
/usr/local/go/src/text/template/exec.go:245
/usr/local/go/src/text/template/exec.go:242
/usr/local/go/src/text/template/exec.go:391
/usr/local/go/src/text/template/exec.go:247
/usr/local/go/src/text/template/exec.go:242
/usr/local/go/src/text/template/exec.go:329
/usr/local/go/src/text/template/exec.go:338
/usr/local/go/src/text/template/exec.go:245
/usr/local/go/src/text/template/exec.go:242
/usr/local/go/src/text/template/exec.go:197
/usr/local/go/src/text/template/exec.go:180
/usr/local/go/src/text/template/exec.go:166
/go/src/github.com/appscode/voyager/pkg/haproxy/template/renderer.go:22
/go/src/github.com/appscode/voyager/pkg/ingress/parser.go:720
/go/src/github.com/appscode/voyager/pkg/ingress/loadbalancer.go:72
/go/src/github.com/appscode/voyager/pkg/operator/ingress_crds.go:103
/go/src/github.com/appscode/voyager/pkg/operator/ingress_crds.go:23
/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:68
/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:51
/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:37
/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: port expected to be > 0, found 0 [recovered]
	panic: port expected to be > 0, found 0 [recovered]
	panic: port expected to be > 0, found 0

goroutine 330 [running]:
github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x111
panic(0x1f54820, 0xc420419480)
	/usr/local/go/src/runtime/panic.go:491 +0x283
text/template.errRecover(0xc4210124e8)
	/usr/local/go/src/text/template/exec.go:146 +0x1c5
panic(0x1f54820, 0xc420419480)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/appscode/voyager/pkg/haproxy/template.HostACLs(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/appscode/voyager/pkg/haproxy/template/template.go:45 +0xcf8
reflect.Value.call(0x20aa340, 0x25195c8, 0x13, 0x2449ebb, 0x4, 0xc42067d7a0, 0x4, 0x4, 0x2434f40, 0x1e9e401, ...)
	/usr/local/go/src/reflect/value.go:434 +0x905
reflect.Value.Call(0x20aa340, 0x25195c8, 0x13, 0xc42067d7a0, 0x4, 0x4, 0x3edd820, 0xc4208b5590, 0x1f27360)
	/usr/local/go/src/reflect/value.go:302 +0xa4
text/template.(*state).evalCall(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0x20aa340, 0x25195c8, 0x13, 0x3edd340, 0xc4208b53e0, 0xc4208b0c60, ...)
	/usr/local/go/src/text/template/exec.go:670 +0x580
text/template.(*state).evalFunction(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0xc4208b5410, 0x3edd340, 0xc4208b53e0, 0xc420461280, 0x5, 0x8, ...)
	/usr/local/go/src/text/template/exec.go:538 +0x176
text/template.(*state).evalCommand(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0xc4208b53e0, 0x0, 0x0, 0x0, 0x1f53de0, 0x0, ...)
	/usr/local/go/src/text/template/exec.go:435 +0x53a
text/template.(*state).evalPipeline(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0xc4206d0370, 0x0, 0x0, 0x3ee7cc0)
	/usr/local/go/src/text/template/exec.go:408 +0x115
text/template.(*state).evalCommand(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0xc4208b53b0, 0x0, 0x0, 0x0, 0x0, 0x3, ...)
	/usr/local/go/src/text/template/exec.go:438 +0x7bf
text/template.(*state).evalPipeline(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0xc4206d0320, 0x0, 0x1ef0b00, 0xc42052fc08)
	/usr/local/go/src/text/template/exec.go:408 +0x115
text/template.(*state).walkIfOrWith(0xc421011ff8, 0x13, 0x1e9e5e0, 0xc42000cc10, 0x196, 0xc4206d0320, 0xc4208b55c0, 0x0)
	/usr/local/go/src/text/template/exec.go:263 +0xbc
text/template.(*state).walk(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0x3edd880, 0xc4208af580)
	/usr/local/go/src/text/template/exec.go:253 +0x2aa
text/template.(*state).walk(0xc421011ff8, 0x1e9e5e0, 0xc42000cc10, 0x196, 0x3edd520, 0xc4208b5320)
	/usr/local/go/src/text/template/exec.go:242 +0x11d
text/template.(*state).walkRange.func1(0x1f53de0, 0x40a9e40, 0x82, 0x1e9e5e0, 0xc42000cc10, 0x196)
	/usr/local/go/src/text/template/exec.go:329 +0x12f
text/template.(*state).walkRange(0xc421011ff8, 0x21094c0, 0xc42000cc28, 0x196, 0xc4208b83c0)
	/usr/local/go/src/text/template/exec.go:338 +0x307
text/template.(*state).walk(0xc421011ff8, 0x21094c0, 0xc42000cc28, 0x196, 0x3edd6a0, 0xc4208b83c0)
	/usr/local/go/src/text/template/exec.go:245 +0x451
text/template.(*state).walk(0xc421011ff8, 0x21094c0, 0xc42000cc28, 0x196, 0x3edd520, 0xc4208ac2a0)
	/usr/local/go/src/text/template/exec.go:242 +0x11d
text/template.(*state).walkTemplate(0xc421012468, 0x21094c0, 0xc42000cc28, 0x196, 0xc42021b300)
	/usr/local/go/src/text/template/exec.go:391 +0x25d
text/template.(*state).walk(0xc421012468, 0x21094c0, 0xc42000cc28, 0x196, 0x3edd760, 0xc42021b300)
	/usr/local/go/src/text/template/exec.go:247 +0x1f6
text/template.(*state).walk(0xc421012468, 0x21094c0, 0xc42000cc28, 0x196, 0x3edd520, 0xc4203e1aa0)
	/usr/local/go/src/text/template/exec.go:242 +0x11d
text/template.(*state).walkRange.func1(0x1f53de0, 0x40a9e40, 0x82, 0x21094c0, 0xc42000cc28, 0x196)
	/usr/local/go/src/text/template/exec.go:329 +0x12f
text/template.(*state).walkRange(0xc421012468, 0x238f480, 0xc42007bb00, 0x99, 0xc42021b440)
	/usr/local/go/src/text/template/exec.go:338 +0x307
text/template.(*state).walk(0xc421012468, 0x238f480, 0xc42007bb00, 0x99, 0x3edd6a0, 0xc42021b440)
	/usr/local/go/src/text/template/exec.go:245 +0x451
text/template.(*state).walk(0xc421012468, 0x238f480, 0xc42007bb00, 0x99, 0x3edd520, 0xc4203e1500)
	/usr/local/go/src/text/template/exec.go:242 +0x11d
text/template.(*Template).execute(0xc42021ac00, 0x3eae2a0, 0xc420f78c40, 0x238f480, 0xc42007bb00, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:197 +0x1f9
text/template.(*Template).Execute(0xc42021ac00, 0x3eae2a0, 0xc420f78c40, 0x238f480, 0xc42007bb00, 0x1, 0xc421012570)
	/usr/local/go/src/text/template/exec.go:180 +0x53
text/template.(*Template).ExecuteTemplate(0xc4208ab2c0, 0x3eae2a0, 0xc420f78c40, 0x2453b1e, 0xb, 0x238f480, 0xc42007bb00, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:166 +0x77
github.com/appscode/voyager/pkg/haproxy/template.RenderConfig(0xc42069ac80, 0xc420796e00, 0x5, 0x8, 0xc4202eff20, 0x2, 0x2, 0x0, 0x40a8d80, 0x0, ...)
	/go/src/github.com/appscode/voyager/pkg/haproxy/template/renderer.go:22 +0x172
github.com/appscode/voyager/pkg/ingress.(*controller).generateConfig(0xc420ab8d80, 0x0, 0x0)
	/go/src/github.com/appscode/voyager/pkg/ingress/parser.go:720 +0x42bf
github.com/appscode/voyager/pkg/ingress.(*loadBalancerController).Reconcile(0xc42000c170, 0x7, 0xc420701fa0)
	/go/src/github.com/appscode/voyager/pkg/ingress/loadbalancer.go:72 +0x5c
github.com/appscode/voyager/pkg/operator.(*Operator).reconcileEngress(0xc42007f880, 0xc42069f080, 0x1a, 0xc42064cda0, 0x6fc600)
	/go/src/github.com/appscode/voyager/pkg/operator/ingress_crds.go:103 +0x859
github.com/appscode/voyager/pkg/operator.(*Operator).(github.com/appscode/voyager/pkg/operator.reconcileEngress)-fm(0xc42069f080, 0x1a, 0xc4204ac480, 0x1f54820)
	/go/src/github.com/appscode/voyager/pkg/operator/ingress_crds.go:23 +0x3e
github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue.(*Worker).processNextEntry(0xc42031ac00, 0x82b700)
	/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:68 +0xeb
github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue.(*Worker).processQueue(0xc42031ac00)
	/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:51 +0x2b
github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue.(*Worker).(github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue.processQueue)-fm()
	/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:37 +0x2a
github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc420419520)
	/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc420419520, 0x3b9aca00, 0x0, 0x1, 0xc42049a4e0)
	/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc420419520, 0x3b9aca00, 0xc42049a4e0)
	/go/src/github.com/appscode/voyager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue.(*Worker).Run
	/go/src/github.com/appscode/voyager/vendor/github.com/appscode/kutil/tools/queue/worker.go:37 +0x90

```

--- a/pkg/ingress/parser.go
+++ b/pkg/ingress/parser.go
@@ -409,22 +409,17 @@ func (c *controller) generateConfig() error {
 	for ri, rule := range c.Ingress.Spec.Rules {
 		if rule.HTTP != nil {
 			binder := hostBinder{Address: rule.HTTP.Address}
-			info := &httpInfo{Hosts: make(map[string][]*hpi.HTTPPath)}
-			if v, ok := httpServices[binder]; ok {
-				info = v
-			} else {
-				httpServices[binder] = info
-			}
+			offloadSSL := false
 
 			if c.Ingress.UseTLSForRule(rule) {
-				info.OffloadSSL = true
+				offloadSSL = true
 				if port := rule.HTTP.Port.IntValue(); port > 0 {
 					binder.Port = port
 				} else {
 					binder.Port = 443
 				}
 			} else {
-				info.OffloadSSL = false
+				offloadSSL = false
 				if port := rule.HTTP.Port.IntValue(); port > 0 {
 					binder.Port = port
 				} else {
@@ -432,6 +427,13 @@ func (c *controller) generateConfig() error {
 				}
 			}
 
+			info := &httpInfo{Hosts: make(map[string][]*hpi.HTTPPath)}
+			if v, ok := httpServices[binder]; ok {
+				info = v
+			} else {
+				httpServices[binder] = info
+			}
+			info.OffloadSSL = offloadSSL
 			info.ALPNOptions = rule.ParseALPNOptions()
 
 			httpPaths := info.Hosts[rule.GetHost()]

